ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -1967,6 +1968,13 @@ private RelNode applyPreJoinOrderingTransforms(RelNode basePlan, RelMetadataProv
       generatePartialProgram(program, false, HepMatchOrder.DEPTH_FIRST,
           HiveExceptRewriteRule.INSTANCE);
 
+      // ?
```

Comment:
We can add a comment here?

Reply:
actually...I was wondering if this is the right place to add this rule or not... :)

Reply:
Good point.
I am not sure the decorrelation logic or any rule executed before this one would introduce a `count distinct` or any of the other functions that we will be targeting.
However, if that would be the case at some point, probably we do not want to be mangling with them in our new rules.
Thus, executing it as the very first step, even before decorrelation, would make sense.